observation/src/main/java/com/linecorp/armeria/client/observation/MicrometerObservationClient.java

```diff
@@ -0,0 +1,134 @@
+
+    /**
+     * Creates a new instance.
+     */
+    private MicrometerObservationClient(HttpClient delegate, ObservationRegistry observationRegistry) {
+        this(delegate, observationRegistry, null);
+    }
+
+    /**
+     * Creates a new instance.
+     */
+    private MicrometerObservationClient(
+            HttpClient delegate, ObservationRegistry observationRegistry,
+            @Nullable HttpClientObservationConvention httpClientObservationConvention) {
+        super(delegate);
+        this.observationRegistry = observationRegistry;
+        this.httpClientObservationConvention = httpClientObservationConvention;
+    }
+
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        final RequestHeadersBuilder newHeaders = req.headers().toBuilder();
+        final HttpClientContext httpClientContext = new HttpClientContext(ctx, newHeaders, req);
+        final Observation observation = HttpClientObservationDocumentation.OBSERVATION.observation(
+                this.httpClientObservationConvention, DefaultHttpClientObservationConvention.INSTANCE,
+                () -> httpClientContext, observationRegistry).start();
+        final HttpRequest newReq = req.withHeaders(newHeaders);
+        ctx.updateRequest(newReq);
+        final RequestContextExtension ctxExtension = ctx.as(RequestContextExtension.class);
+
+        if (!observationRegistry.isNoop() && !observation.isNoop() && ctxExtension != null) {
+            // Make the span the current span and run scope decorators when the ctx is pushed.
+            ctxExtension.hook(observation::openScope);
+        }
+
+        enrichObservation(ctx, httpClientContext, observation);
+
+        return observation.scopedChecked(() -> unwrap().execute(ctx, newReq));
+        // TODO: Maybe we should move the observation stopping here?
+    }
+
+    private void enrichObservation(ClientRequestContext ctx, HttpClientContext httpClientContext,
+                                   Observation observation) {
+        if (observation.isNoop()) {
+            // For no-op spans, we only need to inject into headers and don't set any other attributes.
+            return;
+        }
+
+        ctx.log()
+           .whenAvailable(RequestLogProperty.REQUEST_FIRST_BYTES_TRANSFERRED_TIME)
+           .thenAccept(requestLog -> observation.event(Events.WIRE_SEND));
+
+        ctx.log()
+           .whenAvailable(RequestLogProperty.RESPONSE_FIRST_BYTES_TRANSFERRED_TIME)
+           .thenAccept(requestLog -> {
+               if (requestLog.responseFirstBytesTransferredTimeNanos() != null) {
+                   observation.event(Events.WIRE_RECEIVE);
+               }
+           });
+
+        ctx.log().whenComplete()
+           .thenAccept(requestLog -> {
+               // TODO: ClientConnectionTimings - no hook to be there
+               //  at the moment of those things actually hapening
```

Comment:
I think this information isn't really critical to merge this PR, but still is useful for users going through traces.

Would it be difficult to also provide the option to pass a `timestamp` such that  `observation.event(Event, long)` like brave and otel do? (maybe `DefaultMeterObservationHandler` can just ignore the timestamp)